crates/router/src/core/payments.rs

```diff
@@ -594,7 +594,11 @@ where
         let payment_method_type_list = session_connector_data
             .iter()
             .map(|session_connector_data| session_connector_data.payment_method_type)
-            .collect();
+            .collect::<Vec<_>>();
+        let payment_method_list = session_connector_data
+            .iter()
+            .map(|session_connector_data| session_connector_data.payment_method)
+            .collect::<Vec<_>>();
```

Comment:
Better have this as a `Vec<(PaymentMethod, PaymentMethodType)>` instead of having separate vectors for better associativity.